broker/service/api/v10.py

```diff
@@ -469,9 +469,10 @@ def delete_all_submissions(data):
         ex.UnauthorizedException -- Authetication problem
     """
     check_authorization(data)
+    db_connector.delete_all()
 
-    for id in submissions.keys():
-        delete_submission(id, data)
+    for key in submissions.keys():
```

Comment:
no need to call keys() here.

Reply:
In this case it's necessary because when a submission is deleted, the map's length is decremented.